docs/development/doc_guide.rst

```diff
@@ -1,76 +1,633 @@
-************************
-Documentation Guidelines
-************************
+*******************
+Documentation Guide
+*******************
 
-This document describes the documentation requirements and guidelines
-to be followed during the development of PlasmaPy and affiliated
-packages.
+Documentation that is up-to-date and understandable is vital to the
+health of a software project. This page describes the documentation
+requirements and guidelines to be followed during the development of
+PlasmaPy and affiliated packages.
+
+`PlasmaPy's online documentation`_ is hosted by `Read the Docs`_ and is
+available at these locations.
+
+* The documentation corresponding to the most recent official release
+  is labelled ``stable`` and is found at
+  `https://docs.plasmapy.org/ <https://docs.plasmapy.org/>`_ or
+  `https://docs.plasmapy.org/en/stable/ <https://docs.plasmapy.org/en/stable/>`_.
+
+* The documentation corresponding to the ``main`` branch on
+  `PlasmaPy's GitHub repository`_ is labelled ``latest`` and can be
+  found at `https://docs.plasmapy.org/en/latest/
+  <https://docs.plasmapy.org/en/latest/>`_.
+
+A preview of the documentation is generated every time a pull request
+is created or updated. You can access this preview by scrolling down
+to the checks at the bottom of a pull request, and clicking on
+``Details`` next to ``docs/readthedocs.org:plasmapy``.
 
 Building documentation
 ======================
-Documentation is built from the main branch on every commit pushed
-to it.
 
-Sphinx, the documentation generator of PlasmaPy, uses reStructuredText (reST) as its markup language. A primer on reST is available at this `webpage
+To install all dependencies required to develop PlasmaPy on your local
+computer, enter the top-level directory of the cloned repository and
+run
+
+.. code-block:: bash
+
+   pip install tox -r requirements.txt
+
+You can use `tox`_ to build the documentation from within the main
+PlasmaPy repository directory by running
+
+.. code-block:: bash
+
+   tox -e build_docs
+
+You can access the documentation landing page by opening
+``docs/_build/html/index.html`` with your browser of choice.
+
+When writing documentation, please make sure to fix any warnings that
+arise. To enforce this, the ``build_docs`` environment is set to fail
+on encountering any warnings via the ``-W`` flag to ``sphinx-build``.
+
+You can shorten the documentation build by running
+
+.. code-block:: bash
+
+   tox -e build_docs_no_examples
+
+in order to build the documentation without executing the
+:ref:`example notebooks <example_notebooks>`. This command will pass
+even if there are warnings.  The `tox`_ command may also include the
+``--parallel`` flag to allow a parallel build.
+
+If you have `make <https://www.gnu.org/software/make/>`_ installed,
+then you can build the documentation by entering the ``docs/`` directory
+and running
+
+.. code-block:: bash
+
+   make html
```

Comment:
Make a mention that there is a `make clean` (probably something similar for `tox`) that allows you to remove all build files.  This is often need when you make a change to a file that does not trigger sphinx to rebuild that file you altered.  This happen more frequently than one would expect.